q2_moshpit/busco/plots_summary.py

```diff
@@ -0,0 +1,139 @@
+            width=PLOT_DIM,
+            height=PLOT_DIM
+        )
+        charts.append(chart)
+
+    chart = alt.hconcat(
+        *charts
+    )
+
+    return chart
+
+
+def _draw_marker_summary_histograms(data: pd.DataFrame) -> dict:
+    """
+    Draws summary histograms for the BUSCO marker results of all samples.
+
+    Returns:
+        dict: Dictionary containing the Vega spec.
+    """
+    chart = _draw_horizontal_histograms(
+        data, columns=["single", "duplicated", "fragmented", "missing"]
+    )
+    chart2 = _draw_horizontal_histograms(
+        data, columns=["scaffolds", "contigs_n50", "scaffold_n50", "length"]
+    )
+
+    chart = alt.vconcat(chart, chart2).configure_axis(
+        labelFontSize=LABEL_FONT_SIZE, titleFontSize=TITLE_FONT_SIZE
+    ).configure_legend(
+        labelFontSize=LABEL_FONT_SIZE, titleFontSize=TITLE_FONT_SIZE
+    ).configure_header(
+        labelFontSize=LABEL_FONT_SIZE, titleFontSize=TITLE_FONT_SIZE
+    )
+
+    return chart.to_dict()
+
+
+def _draw_selectable_summary_histograms(data: pd.DataFrame) -> dict:
+    """
+    Draws summary histograms for the MAG assembly metrics where users
+    can indicate which metric and for which sample they want to see.
+
+    Returns:
+        dict: Dictionary containing the Vega spec.
+    """
+    metrics = [
+        'single', 'duplicated', 'fragmented', 'missing',
+        'scaffolds', 'contigs_n50', 'length'
+    ]
+    data = pd.melt(
+        data,
+        id_vars=["sample_id", "mag_id", "dataset", "n_markers"],
+        value_vars=metrics,
+        value_name="metric",
+        var_name="category",
+    )
+
+    # Create the dropdown selection with all possible metrics
+    selection_metrics = alt.selection_point(
+        fields=['category'],
+        bind=alt.binding_select(options=metrics),
+        name='select_metric',
+        value='single'
+    )
+
+    # Create the sample search box
+    samples = data['sample_id'].unique().tolist()
+    selection_box = alt.param(
+        value=samples[0],
+        name='select_sample',
+        bind=alt.binding(
+            input='search',
+            placeholder="Sample IDs",
+            name='select_sample',
+        )
+    )
+
+    # Create the chart
+    chart = alt.Chart(data).mark_bar().encode(
+        x=alt.X('metric:Q', bin=True, title=None),
+        y=alt.Y('count()', title='MAG count'),
+        color=alt.value("steelblue")
+    ).add_params(
+        selection_metrics,
+        selection_box
+    ).transform_filter(
+        'datum.sample_id == trim(select_sample) '
+        '& datum.category == select_metric.category'
+    ).configure_axis(
+        labelFontSize=LABEL_FONT_SIZE, titleFontSize=TITLE_FONT_SIZE
+    ).configure_legend(
+        labelFontSize=LABEL_FONT_SIZE, titleFontSize=TITLE_FONT_SIZE
+    ).configure_header(
+        labelFontSize=LABEL_FONT_SIZE, titleFontSize=TITLE_FONT_SIZE
+    ).properties(
+        width=PLOT_DIM,
+        height=PLOT_DIM
+    )
```

Comment:
![Screenshot 2024-04-25 at 11 38 35](https://github.com/bokulich-lab/q2-moshpit/assets/54123712/3d838630-ad99-4866-a4f9-c44828741361)

In the same vein as the suggestion above, I think it would improve comparability if the domain of the per-sample histogram is the same as that of the all-samples histogram. This might be a little more annoying to implement.

Reply:
That is a cool suggestion, thanks - I'm going to create a new issue for that (after this is merged), though, as it will require some looking into, as you noticed yourself.

Reply:
Just to add information to this... I just noticed that for one of the inputs I used to test the action, I get histograms with only one bin. 
![Screenshot 2024-04-25 at 17 30 18](https://github.com/bokulich-lab/q2-moshpit/assets/54123712/01b2686a-15cf-4a64-b027-077585b44efa)

This is the input artifact and the command I used if you want to try it.
[w9-mags.qza.zip](https://github.com/bokulich-lab/q2-moshpit/files/15112442/w9-mags.qza.zip)
```bash
qiime moshpit evaluate-busco --i-bins w9-mags.qza --o-visualization w9-mags.qzv --p-lineage-dataset bacteria_odb10 --verbose --o-results-table results.qza --parallel --p-num-partitions 4
```